en/news/_posts/2017-12-25-ruby-2-5-0-released.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: news_post
+title: "Ruby 2.5.0 Released"
+author: "naruse"
+translator:
+date: 2017-12-25 00:00:00 +0000
+lang: en
+---
+
+We are pleased to announce the release of Ruby 2.5.0.
+
+Ruby 2.5.0 is the first stable release of Ruby 2.5 series.
+It introduces many new features and performance improvements, for example:
+
+## New Features
+
+* Print backtrace and error message in reverse order if STDERR is unchanged and a tty.
+  [Feature #8661] [experimental]
+
+* Top-level constant look-up is removed.  [Feature #11547]
+
+* rescue/else/ensure are allowed inside do/end blocks.  [Feature #12906]
+
+* Add yield\_self.  [Feature #6721]
+
+## Performance improvements
+
+* Dynamic instrumentation for TracePoint hooks instead of using "trace"
```

Comment:
About 10% speed-up by removing all `trace` instructions from overall bytecodes (instruction sequences). `trace` instructions were inserted to support `TracePoint`. However, most of case `TracePoint` is not used and `trace` instructions are pure-overhead. Instead of `trace` instruction, we use dynamic instrumentation technique. See [Feature #14104] for details.